README.md

```diff
@@ -133,166 +133,215 @@ The travel agency service checks if all three airlines available in parallel and
     }
 
     // Worker to communicate with airline 'Emirates'
-    worker emiratesWorker {
-        http:Request outReq;
+    worker emiratesWorker returns http:Response? {
+        http:Request outReq = new;
         // Out request payload
         outReq.setJsonPayload(untaint flightPayload);
         // Send a POST request to 'Emirates' and get the results
-        http:Response respWorkerEmirates = check airlineEP -> post("/emirates",
-            outReq);
-        // Reply to the join block - Send the response from 'Emirates'
-        respWorkerEmirates -> fork;
+        var respWorkerEmirates = airlineEP->post("/emirates", outReq);
+        // Reply to the join block from this worker - Send the response from 'Emirates'
+        if (respWorkerEmirates is http:Response) {
+            return respWorkerEmirates;
+        }
+        return;
     }
-} join (all) (map airlineResponses) {
-    // Wait until the responses received from all the workers running in parallel
-    int qatarPrice;
-    int asianaPrice;
-    int emiratesPrice;
-
-    // Get the response and price for airline 'Qatar Airways'
-    var resQatar = check <http:Response>(airlineResponses["qatarWorker"]);
-    jsonFlightResponseQatar = check resQatar.getJsonPayload();
-    qatarPrice = jsonFlightResponseQatar.Price but {
-        int price => price,
-        any otherVals => -1
-    };
-
-    // Get the response and price for airline 'Asiana'
-    var resAsiana = check <http:Response>(airlineResponses["asianaWorker"]);
-    jsonFlightResponseAsiana = check resAsiana.getJsonPayload();
-    asianaPrice = jsonFlightResponseAsiana.Price but {
-        int price => price,
-        any otherVals => -1
-    };
-
-    // Get the response and price for airline 'Emirates'
-    var resEmirates = check <http:Response>(airlineResponses["emiratesWorker"]);
-    jsonFlightResponseEmirates = check resEmirates.getJsonPayload();
-    emiratesPrice = jsonFlightResponseEmirates.Price but {
-        int price => price,
-        any otherVals => -1
-    };
-
-    // Select the airline with the least price
-    if (qatarPrice < asianaPrice) {
-        if (qatarPrice < emiratesPrice) {
-            jsonFlightResponse = jsonFlightResponseQatar;
+}
+
+// Wait until the responses received from all the workers running in parallel
+record{
+    http:Response? qatarWorker;
+    http:Response? asianaWorker;
+    http:Response? emiratesWorker;
+} airlineResponses = wait {qatarWorker, asianaWorker, emiratesWorker};
+
+int qatarPrice = -1;
+int asianaPrice = -1;
+int emiratesPrice = -1;
+
+// Get the response and price for airline 'Qatar Airways'
+var resQatar = airlineResponses["qatarWorker"];
+if (resQatar is http:Response) {
+    var flightResponseQutar= resQatar.getJsonPayload();
+    if (flightResponseQutar is json) {
+        jsonFlightResponseQatar = flightResponseQutar;
+        var qutarResult = jsonFlightResponseQatar.Price;
+        if (qutarResult is int) {
+            qatarPrice = qutarResult;
```

Comment:
There is spelling mismatch. I think it should be `qatar`

Reply:
+1